protos/transaction.proto

```diff
@@ -39,8 +39,8 @@ message TransactionHeader {
     // what addresses the transaction processor is allowed to read from.
     repeated string inputs = 5;
 
-    // A random string that will allow the same transaction to be submitted
-    // several times
+    // A random string that will allow transactions have the same fields
```

Comment:
How about: // A random string that provides uniqueness for transactions with otherwise identical fields.